### PR TITLE
Add order type application table

### PR DESCRIPTION
**Pull request by islean**
## Description

First part of https://github.com/Clinical-Genomics/add-new-tech/issues/123. Adds a new table for mapping order types to their allowed applications

### Added

- OrderTypeApplication table

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b add-ordertype-application-table -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by islean** [comment]
For some reason the order types are added in all caps, will that be an issue?: 

image

File: cg/store/models.py [hunk 1]
```diff
@@ -1189,3 +1191,15 @@ class PacbioSampleSequencingMetrics(SampleRunMetrics):
...
+
+class OrderTypeApplication(Base):
+    """Maps an order type to its allowed applications"""
+
+    __tablename__ = "order_type_application"
+
+    order_type: Mapped[OrderType] = mapped_column(sqlalchemy.Enum(OrderType), primary_key=True)
+    application_id: Mapped[int] = mapped_column(
+        ForeignKey("application.id", ondelete="CASCADE"), primary_key=True
+    )
```
Comment (ChrOertlin) [inline cg/store/models.py:1204]: will this delete the application when deleted? Or is this deleted when the application is removed from the application table?

**Review by ChrOertlin — APPROVED** [review]
Looks good, just a question about the cascade direction

File: cg/store/models.py (hunk 1 above)
Comment (islean) [reply cg/store/models.py:1204]: Deleting the application will delete entries in OrderTypeApplication but not the other way around!

**Comment by Vince-janv** [comment]
> For some reason the order types are added in all caps, will that be an issue?:

This is what most libraries do when they encounter Enums (remember the long Enum debate 😅 ). If we want to stay with lower case in the database we can use [this option](https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.Enum.params.values_callable) to still get the correct typing

**Comment by islean** [comment]
Deployed to hasta:

```
/home/isak.ohlsson
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade 3b165ec34791 -> fd7c9d246255, Add ordertype application table
[isak.ohlsson@hasta:~] [P_base] $
```

**Pull request merged**